server/api.go

```diff
@@ -0,0 +1,243 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+const externalAPICacheTTL = 3600000
+
+var externalAPICache []byte
+var externalAPILastUpdate int64
+var externalAPICacheMutex sync.Mutex
+
+type StartMeetingRequest struct {
+	ChannelID string `json:"channel_id"`
+	Personal  bool   `json:"personal"`
+	Topic     string `json:"topic"`
+	MeetingID int    `json:"meeting_id"`
+}
+
+type StartMeetingFromAction struct {
+	model.PostActionIntegrationRequest
+	Context struct {
+		MeetingID    string `json:"meeting_id"`
+		MeetingTopic string `json:"meeting_topic"`
+		Personal     bool   `json:"personal"`
+	} `json:"context"`
+}
+
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	switch path := r.URL.Path; path {
+	case "/api/v1/meetings/enrich":
+		p.handleEnrichMeetingJwt(w, r)
+	case "/api/v1/meetings":
+		p.handleStartMeeting(w, r)
+	case "/api/v1/config":
+		p.handleConfig(w, r)
+	case "/jitsi_meet_external_api.js":
+		p.proxyExternalAPIjs(w, r)
+	default:
+		http.NotFound(w, r)
+	}
+}
+
+func (p *Plugin) handleConfig(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-Id")
+
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	config, err := p.getUserConfig(userID)
+	if err != nil {
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+
+	b, err2 := json.Marshal(config)
+	if err2 != nil {
+		log.Printf("Error marshaling the Config to json: %v", err2)
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_, _ = w.Write(b)
```

Comment:
I prefer a small quantity of repeated straight forward code, than a not-so-needed abstraction. We can add that abstraction there, but we have just a bunch of places where we use it, and is crystal clear what is doing now, if we add the abstraction we need to read the `WriteResponse` function to understand it.

Reply:
I prefer the abstraction because that way "we do not forget" about doing certain stuff, and we make sure it is done always the same way. Nevertheless, I also see your point. I leave the final decision to you.

Reply:
Done, look at the last added commit.